File: 백준/Silver/2312. 수 복원하기/수 복원하기.py
n = int(input())
for i in range(n):
    num = int(input())

    for j in range(2, num + 1):
        cnt = 0
        if num % j == 0:
            while num % j == 0:
                num //= j
                cnt += 1
            print(j, cnt)
        elif num == 1:
            break

File: 백준/Silver/2312. 수 복원하기/수 복원하기.cc
#include <bits/stdc++.h>

using namespace std;

int main() {
    int n;
    cin >> n;
    for (int i = 0; i < n; i++) {
        int num;
        cin >> num;
        for (int j = 2; j < num + 1; j++) {
            int cnt = 0;
            if (num % j == 0) {
                while (num % j == 0) {
                    num /= j;
                    cnt += 1;
                }
                cout << j << ' ' << cnt << '\n';
            } else if (num == 1) {
                break;
            }
        }
    }
    return 0;
}